components/prism-odin.js

```diff
@@ -0,0 +1,61 @@
+(function (Prism) {
+	var escapes = /\\(?:["'\\abefnrtv]|0[0-7]{2}|U[\dA-Fa-f]{6}|u[\dA-Fa-f]{4}|x[\dA-Fa-f]{2})/.source;
+
+	function withEscapes(pattern, flags) {
+		return RegExp(pattern.replace(/<escapes>/g, escapes), flags);
+	}
+
+	Prism.languages.odin = {
+		/**
+		 * The current implementation of multiline comments doesn't support nesting.
+		 */
+		'comment': [
+			{
+				pattern: /\/\*[\s\S]*?(?:\*\/|$)/,
+				greedy: true
+			},
+			{
+				pattern: /#![^\n\r]*/,
+				greedy: true
+			},
+			{
+				pattern: /\/\/[^\n\r]*/,
+				greedy: true
+			}
+		],
+
+		/**
+		 * Should be found before strings because of '"'"- and '`'`-like sequences.
+		 */
+		'char': {
+			pattern: withEscapes(/'(?:<escapes>|[^\n\r'\\])'/.source),
+			inside: {
+				'symbol': withEscapes(/<escapes>/.source)
+			}
+		},
+
+		'string': [
+			{
+				pattern: /`[^`]*`/,
+				greedy: true
+			},
+			{
+				pattern: withEscapes(/"(?:<escapes>|[^\n\r"\\])*"/.source),
+				greedy: true,
+				inside: {
+					'symbol': withEscapes(/<escapes>/.source)
+				}
+			}
+		],
+
+		'number': /(?:\b0(?:b[01_]+|d[\d_]+|h_*(?:(?:(?:[\dA-Fa-f]_*){8}){1,2}|(?:[\dA-Fa-f]_*){4})|o[0-7_]+|x[\dA-F_a-f]+|z[\dAB_ab]+)\b|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:[Ee][+-]?\d*)?[ijk]?)/,
+
+		'boolean': /\b(?:_|false|nil|true)\b/,
```

Comment:
I still think that `_` should be a keyword. It's a discard operator, right?

Reply:
It is a discard, er... _thing_ :) It seems to me that `_` behaves like a _write-only variable_. Both in Odin and C# you can “assign” a value to the discard variable, but you can’t get its value, because it doesn’t have any value. The C# guide [defines](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards) “discards” as variables.

```odin
package main

import "core:fmt"

main :: proc() {
  _ = 42 // Works!
  fmt.println(_) // '_' cannot be used as a value
}
```

Note that the approach to retrieve the value of `_` is a compile-time error, but it is not named a “syntax error”. By contrast, if you use a keyword or an operator as a left-side operand in an assignment expression, Odin shows a **syntax** error message.

Reply:
The redeclaration of `_` causes a compile-time error, but the error message doesn’t have a syntax error label.

```odin
package main

main :: proc() {
  _: int // No new declarations on the lhs
}
```

The declaration of variables called `+` and `return` causes a compile-time error. The error message has a syntax error label:

```odin
package main

main :: proc() {
  +: int      // Syntax Error: Expected an operand
  return: int // Syntax Error: Expected an operand
}
```

Reply:
Hmmm, I'm still in favor of making this a keyword, or maybe an operator. It's just that `boolean` is definitely not fitting. I'm somewhat against `variable` because this token name is typically reserved for variables you can read from.

Reply:
Yes, `variable` would look especially weird because no other variables are highlighted (we can’t distinguish between variables and constants).

Reply:
I’ve made it a keyword. See 7a292f2.